src/renderer/components/SignMessageConfirm/index.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import invariant from "invariant";
+import React from "react";
+import { withTranslation } from "react-i18next";
+import type { TFunction } from "react-i18next";
+import styled from "styled-components";
+import type { AccountLike } from "@ledgerhq/live-common/lib/types";
+import type { TypedMessageData } from "@ledgerhq/live-common/lib/families/ethereum/types";
+import type { MessageData } from "@ledgerhq/live-common/lib/hw/signMessage/types";
+import type { DeviceTransactionField } from "@ledgerhq/live-common/lib/transaction";
+import type { Device } from "@ledgerhq/live-common/lib/hw/actions/types";
+import Box from "~/renderer/components/Box";
+import Text from "~/renderer/components/Text";
+import useTheme from "~/renderer/hooks/useTheme";
+import { renderVerifyUnwrapped } from "~/renderer/components/DeviceAction/rendering";
+import SignMessageConfirmField from "./SignMessageConfirmField";
+
+const FieldText = styled(Text).attrs(() => ({
+  ml: 1,
+  ff: "Inter|Medium",
+  color: "palette.text.shade80",
+  fontSize: 3,
+}))`
+  word-break: break-all;
+  text-align: right;
+  max-width: 50%;
+`;
+
+export type FieldComponentProps = {
+  account: AccountLike,
+  field: DeviceTransactionField,
+};
+
+export type FieldComponent = React$ComponentType<FieldComponentProps>;
+
+const TextField = ({ field }: FieldComponentProps) => {
+  invariant(field.type === "text", "TextField invalid");
+  return (
+    <SignMessageConfirmField label={field.label}>
+      <FieldText>{field.value}</FieldText>
+    </SignMessageConfirmField>
+  );
+};
+
+const Container = styled(Box).attrs(() => ({
+  alignItems: "center",
+  fontSize: 4,
+  pb: 4,
+}))``;
+
+type Props = {
+  t: TFunction,
+  device: Device,
+  account: AccountLike,
+  signMessageRequested: TypedMessageData | MessageData,
+};
+
+const SignMessageConfirm = ({ t, device, account, signMessageRequested: message }: Props) => {
+  const type = useTheme("colors.palette.type");
+
+  if (!device) return null;
+
+  const fields = [];
+
+  if (message.hashes && message.hashes.domainHash) {
+    fields.push({
+      type: "text",
+      label: t("SignMessageConfirm.domainHash"),
+      // $FlowFixMe
```

Comment:
😭  what's the best way to type guard in flow?

Reply:
no se :D I did the if above so i dont get why flow does not understand